src/app/scouter/page.tsx

```diff
@@ -0,0 +1,91 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+import { ScouterViewer } from '@/components/ui/ScouterViewer';
+import { Button, createButtonProps } from '@/components/ui/Button';
+import { useState, useEffect } from 'react';
+import { getBattlePower } from '@/app/calc/ScouterCalculator';
+
+// 戦闘力計算用コンポーネント
+function ResultButton({ follower }: { follower: number }) {
+  const router = useRouter();
+  
+  const handleSubmit = () => {
+    const battlePower = getBattlePower(follower);
+    sessionStorage.setItem('battlePower', String(battlePower));
+    router.push('/result');
+  };
+  
+  return (
+    <div className="mt-2">
+      <Button
+        button={createButtonProps('button', '結果を見る')}
+        onClick={handleSubmit}
+      />
+    </div>
+  );
+}
+
+// ScouterViewerを表示するコンポーネント
+function ScouterSection({ onRotationComplete }: { onRotationComplete: () => void }) {
+  return (
+    <ScouterViewer onRotationComplete={onRotationComplete} />
+  );
+}
+
+export default function ScouterPage() {
+  const router = useRouter();
+  const [followerNumber, setFollowerNumber] = useState<number | null>(null);
+  const [rotationCompleted, setRotationCompleted] = useState(false);
+  const [isRedirecting, setIsRedirecting] = useState(false);
+
+  // セッションストレージからフォロワー数を取得
+  useEffect(() => {
+    const storedFollower = sessionStorage.getItem('follower');
+    if (!storedFollower) {
+      setIsRedirecting(true);
+      return;
+    }
+
+    const numFollower = Number(storedFollower);
+    if (isNaN(numFollower)) {
+      setIsRedirecting(true);
+      return;
+    }
+
+    setFollowerNumber(numFollower);
+  }, []);
```

Comment:
セッションストレージの値は、不要になったら小まめに削除しておきましょう
※もしリザルト画面でセッションストレージからの値取得に失敗した時、再度このfollowerからの取得を試みる、などのハンドリングをする場合はこの限りではありませんmm

```suggestion
  // セッションストレージからフォロワー数を取得
  useEffect(() => {
    const storedFollower = sessionStorage.getItem('follower');
    if (!storedFollower) {
      setIsRedirecting(true);
      return;
    }

    const numFollower = Number(storedFollower);
    if (isNaN(numFollower)) {
      setIsRedirecting(true);
      return;
    }

    setFollowerNumber(numFollower);
    sessionStorage.removeItem('follower');
  }, []);
```

Reply:
useEffectの開発者モードが悪さしてるっぽく、コンポーネントが呼び出されたときに一度削除されて、その後もう一度呼び出されたときにはセッションがないことから、storedFollowerがnullとなり、storedFollowの条件がtrueとなってるみたいで、リダイレクトしてしまうので、セッション情報の削除処理は一旦無くしました😭